classes/class-p4-taxonomy-page.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+if ( ! class_exists( 'P4_Taxonomy_Page' ) ) {
+	/**
+	 * Class P4MT_Taxonomy_Page
+	 */
+	class P4_Taxonomy_Page {
+
+		/**
+		 * Taxonomy_Page constructor.
+		 */
+		public function __construct() {
+			$this->hooks();
+		}
+
+		/**
+		 * Class hooks.
+		 */
+		private function hooks() {
+			add_action( 'category_add_form_fields',              [ $this, 'add_taxonomy_form_fields' ] );
+			add_action( 'category_edit_form_fields',             [ $this, 'add_taxonomy_form_fields' ] );
+			add_action( 'edited_category',                       [ $this, 'save_taxonomy_meta' ] );
+			add_action( 'create_category',                       [ $this, 'save_taxonomy_meta' ] );
+
+			add_filter( 'manage_edit-category_columns',          [ $this, 'edit_taxonomy_columns' ] );
+			add_filter( 'manage_category_custom_column',         [ $this, 'manage_taxonomy_custom_column' ], 10, 3 );
+			add_filter( 'manage_edit-category_sortable_columns', [ $this, 'manage_taxonomy_custom_sortable_column' ], 10, 3 );
+		}
+
+		/**
+		 * Add custom field(s) to taxonomy form.
+		 *
+		 * @param WP_Term $wp_tag The object passed to the callback when on Edit Tag page.
+		 */
+		public function add_taxonomy_form_fields( $wp_tag ) {
+
+			$args = [
+				'post_type'        => 'page',
+				'show_option_none' => 'None',
+				'name'             => 'category_page',
+				'id'               => 'category-page-id',
+				'class'            => 'category-page-class',
+			];
+
+			// Filter issue pages.
+			$options          = get_option( 'planet4_options' );
+			$explore_page_id  = $options['explore_page'];
+			if ( $explore_page_id ) {
+				$args['child_of'] = $explore_page_id;
+			}
+
+			$args = array_map( 'esc_attr', $args );
+
+			if ( isset( $wp_tag ) && $wp_tag instanceof WP_Term ) {
+				$category_page = get_term_meta( $wp_tag->term_id, 'category_page', true );
+				if ( $category_page ) {
+					$args['selected'] = $category_page;
+				} ?>
+				<tr class="form-field edit-wrap term-category-page-wrap">
+					<th>
+						<label><?php echo esc_html__( 'Select Category Page', 'planet4-master-theme' ); ?></label>
+					</th>
+					<td>
+						<?php wp_dropdown_pages( $args ); ?>
```

Comment:
@sagarsdeshmukh 
Actually, could you do this right when you do output (lines 64, 71) because CS is still complaining about not
escaping output.
So, it could be `wp_dropdown_pages( $array_map( 'esc_attr', $args ) );`